third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -615,6 +616,18 @@ func isFirstGen(d *schema.ResourceData) bool {
 	return !regexp.MustCompile("db*").Match([]byte(tier))
 }
 
+// Makes private_network ForceNew if it is changing from set to nil. The API returns an error
+// if this change is attempted in-place.
+func privateNetworkCustomizeDiff(d *schema.ResourceDiff, meta interface{}) error {
+	old, new := d.GetChange("settings.0.ip_configuration.0.private_network")
+
+	if old != "" && new == "" {
```

Comment:
We'd normally need to check if the new value was known or unknown, but it's unlikely that it would be unknown given that this is a block (and the field itself isn't one we expect to be interpolated).